doc/algorithms/quantum_neural_network.rst

```diff
@@ -0,0 +1,170 @@
+
+Quantum neural networks aim to encode neural networks into a quantum system, with the intention of benefitting from quantum information processing. There have been numerous attempts to define a quantum neural network, each with varying advantages and disadvantages. The quantum neural network detailed below, following the work of :cite:`killoran2018continuous`, has a CV architecture and is naturally realized using standard CV gates from Strawberry Fields.
+
+CV implementation
+------------------------------------
+
+A CV quantum neural network layer can be defined as 
+
+.. math:: \mathcal{L} := \Phi \circ \mathcal{D} \circ \mathcal{U}_{2} \circ \mathcal{S} \circ \mathcal{U}_{1},
+
+where :math:`\mathcal{U}_{k}=U_{k}(\boldsymbol{\theta}_{k},\boldsymbol{\phi}_{k})` is an :math:`N` mode interferometer, :math:`\mathcal{D}=\otimes_{i=1}^{N}D(\alpha_{i})` and :math:`\mathcal{S}=\otimes_{i=1}^{N}S(r_{i})` are displacement gates (:class:`~.Dgate`) and squeezing gates (:class:`~.Sgate`) acting on each mode with :math:`\alpha_{i} \in \mathbb{C}` and :math:`r_{i} \in \mathbb{R}`, and finally :math:`\Phi=\otimes_{i=1}^{N}\Phi(\lambda_{i})` is a non-Gaussian gate on each mode with parameter :math:`\lambda_{i} \in \mathbb{R}`. Any non-Gaussian gate such as the cubic phase gate (:class:`~.Vgate`) represents a valid choice, but we recommend the Kerr gate (:class:`~.Kgate`) for simulations in Strawberry Fields as it is diagonal in the Fock basis. The layer is shown below as a circuit:
+
+:html:`<br>`
+
+.. image:: ../_static/layer.svg
+    :align: center
+    :width: 70%
+    :target: javascript:void(0);
+
+:html:`<br>`
+
+These layers can then be composed to form a quantum neural network (see :cite:`killoran2018continuous` to understand how to vary the network width).
+
+Let's see how the quantum layer can embed the transformation :math:`\mathcal{L}(\mathbf{x}) = \varphi (W \mathbf{x} + \mathbf{b})` of a classical neural network layer. Suppose :math:`N`-dimensional data is encoded in position eigenstates so that
+
+.. math:: \mathbf{x} \Leftrightarrow \ket{\mathbf{x}} := \ket{x_{1}} \otimes \ldots \otimes \ket{x_{N}}.
+
+We want to perform the transformation
+
+.. math:: \ket{\mathbf{x}} \Rightarrow \ket{\varphi (W \mathbf{x} + \mathbf{b})}.
+
+It turns out that the quantum circuit above can do precisely this! Consider first the affine transformation :math:`W \mathbf{x} + \mathbf{b}`. We can write :math:`W = O_{2} \Sigma O_{1}` with :math:`O_{k}` orthogonal matrices and :math:`\Sigma` a positive diagonal matrix. These orthogonal transformations can be carried out using interferometers without access to phase, i.e., with :math:`\boldsymbol{\phi}_{k} = 0`:

```

Comment:
Might also be good to provide a hyperlink somewhere here to the Bloch-Messiah decomposition: https://strawberryfields.readthedocs.io/en/latest/conventions/decompositions.html#bloch-messiah-or-euler-decomposition

```suggestion
It turns out that the quantum circuit above can do precisely this! Consider first the affine transformation :math:`W \mathbf{x} + \mathbf{b}`. We can always write :math:`W = O_{2} \Sigma O_{1}` with :math:`O_{k}` orthogonal matrices and :math:`\Sigma` a positive diagonal matrix. These orthogonal transformations can be carried out using interferometers without access to phase, i.e., with :math:`\boldsymbol{\phi}_{k} = 0`:
```

Reply:
Ok will do!